### PR TITLE
mgr/rook: pass zone attribute to CephObjectStore CR when creating rgw

### PR DESCRIPTION
**Pull request by josephsawaya**
Added the option to pass zone when running `ceph orch apply rgw <rgw-name>` with the rook orchestrator backend using `--realm=<realm-name>` and `--zone=<zone-name>` options. 

Pointed the rook-client-python submodule at [Sebastian Wagner's repo](https://github.com/sebastian-philipp/rook-client-python/tree/update-june-21) which is updated with the correct API classes that the mgr/rook code is using.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -58,7 +58,8 @@
 	url = https://github.com/ceph/spawn.git
 [submodule "src/pybind/mgr/rook/rook-client-python"]
 	path = src/pybind/mgr/rook/rook-client-python
-	url = https://github.com/ceph/rook-client-python.git
+	url = https://github.com/sebastian-philipp/rook-client-python.git
```
Comment (tchaikov) [inline .gitmodules:61]: @sebastian-philipp @josephsawaya does it make sense to update ceph/rook-client-python to be in sync with sebastian-philipp/rook-client-python ?

Comment (sebastian-philipp) [reply .gitmodules:61]: There is one bit of information missing in the PR description: We're waiting for https://github.com/rook/rook-client-python/pull/4 to get merged. using my fork is just a stop-gap, till this PR is merged, which is blocked by https://github.com/rook/rook/pull/8122

Comment (sebastian-philipp) [reply .gitmodules:61]: PR is merged. we can move over to rook/rook-client-python now

Comment (josephsawaya) [reply .gitmodules]: I've moved it over to [rook/rook-client-python](https://github.com/rook/rook-client-python) in my latest commit

**Pull request merged**